pkg/tfbridge/provider.go

```diff
@@ -977,8 +992,11 @@ func (p *Provider) Update(ctx context.Context, req *pulumirpc.UpdateRequest) (*p
 	// ignoring changes to the keys that would result in replacement/deletion.
 	doIgnoreChanges(ctx, res.TF.Schema(), res.Schema.Fields, olds, news, req.GetIgnoreChanges(), diff)
 
-	contract.Assertf(!diff.Destroy() && !diff.RequiresNew(),
-		"Expected diff to not require deletion or replacement during Update of %s", urn)
+	if diff.Destroy() || diff.RequiresNew() {
+		return nil, fmt.Errorf("internal: expected diff to not require deletion or replacement"+
+			" during Update of %s. Found delete=%t, replace=%t. This indicates a bug in provider.",
+			urn, diff.Destroy(), diff.RequiresNew())
```

Comment:
We know this can happen, so we shouldn't panic here. Instead we can return a helpful error here.

Reply:
Will it still happen after the change here?

Reply:
It *shouldn't*, but the assert *shouldn't* have happened either.